packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -94,25 +87,39 @@ const ShowHide = (props) => {
 
   const theme = React.useContext(ThemeContext);
   const contentRef = React.useRef(null);
+  const [activeElement, setActiveElement] = React.useState(null);
+  const focusRefClassName = 'show-hide-ref-focusable';
 
   /**
-   * Upon showing hidden content, set focus to the child element if focusRef provided, othervise set focus to the content container.
-   * That would allow assistive technologies to read the newly revealed content.
+   * Upon showing hidden content, set activeElement to focusRef element if provided, othervise set it to content container.
    */
   React.useEffect(() => {
+    let newActiveElement = null;
     if (isOpen) {
-      let element = null;
       if (focusRef?.current) {
-        element = focusRef?.current;
+        if (checkIfAddTabindex(focusRef?.current)) {
+          // not focusable element with no tabindex set, safe to temporary add tabindex that will later be removed on blur
+          focusRef?.current.setAttribute('tabIndex', '-1');
+          focusRef?.current.setAttribute('role', 'group');
+          focusRef?.current.classList.add(focusRefClassName); // needed for onBlur cleanup
+        }
+        newActiveElement = focusRef?.current;
       } else if (contentRef?.current) {
-        element = contentRef?.current;
-      }
-      if (element) {
-        focusElement(element);
+        newActiveElement = contentRef?.current;
       }
     }
+    setActiveElement(newActiveElement);
   }, [isOpen, focusRef, contentRef]);
 
+  /**
+  * Set focus to active element allows assistive technologies to read the newly revealed content.
+  */
+  React.useEffect(() => {
+    if (activeElement && isOpen) {
+      activeElement.focus();
+    }
+  }, [activeElement, isOpen]);
+
```

Comment:
I don't believe you should need this additional useEffect hook.

Reply:
I need to set focus to the actual active element once it changes, I don't see how it can be done without the useEffect monitoring that change.

Reply:
Can't you set focus in the hook that is triggered by the isOpen prop change?

Reply:
I wouldn't work because that would trigger focus before tabIndex change to -1 in contentRef, meaning setting focus to not focusable element.